src/database/database.cc

```diff
@@ -2179,6 +2179,13 @@ error Database::saveModel(
                           useRef(model->LocalID()),
                           now;
             } else {
+                // For cases where we need to remove the ID from the TE (like when changing the workspace)
+                *session_ <<
+                          "update time_entries set "
```

Comment:
why is this a separate query and not included into the subsequent query which is also an `update` query with the same `where` condition?

Reply:
I wanted it to stand out more in case something is not clear in the future. Considering how the update looks below, it felt like it would be easy to confuse things.
It could be inside too of course, I can move it if you'd like it better that way.

Reply:
This is certainly more readable and clear, but the reason I mentioned this is performance, 2 requests are likely slower than one. If we know there is no noticeable difference when doing this on `time_entries` table with 1000-2000 rows then this is fine by me.

Reply:
This is definitely more of a special case. This is executed only when you move a TE from one workspace to another from the desktop client. Cases when this happens for more than one TE at a time are very rare, and I'd say even a chance for this happening for 2 TEs at the same time is next to none. 
Anyway, I don't really have a strong opinion where to have this and for any potential future features, it's better to do it in one query. I'll change it.